enhancements/security/bmc-ca-certificate-support.md

```diff
@@ -0,0 +1,294 @@
+  initial CA certificate configuration:
+  1. The cluster administrator creates a ConfigMap named `bmc-verify-ca` in
+     `openshift-machine-api` namespace, containing the desired CA certificates.
+  2. The cluster administrator confirms that the `bmc-verify-ca` ConfigMap has
+     been mounted to `/certs/ca/bmc` in Ironic container.
+  3. The cluster administrator ensures that `disableCertificateVerification`
+     is not set to true in BareMetalHost objects.
+
+### API Extensions
+
+Add an optional map field `bmcCACert` under `platform.baremetal` in
+`install-config.yaml`. Users can configure multiple sets of CA certificates
+with the names they desire, and within each set, multiple certificates can
+also be configured. For example, setting a generic CA certificate (which
+includes multiple certificates) for the cluster and assigning dedicated CA
+certificates to certain baremetal nodes.
+
+```diff
+ # install-config.yaml
+ platform:
+   baremetal:
+     apiVIP: xxx.xxx.xxx.xxx
+     ingressVIP: xxx.xxx.xxx.xxx
+     provisioningNetworkCIDR: xxx.xxx.xxx.xxx/xx
++    bmcCACert:
++      "generic.crt": |
++      -----BEGIN CERTIFICATE-----
++      ......
++      ......
++      ......
++      -----END CERTIFICATE-----
++      -----BEGIN CERTIFICATE-----
++      ......
++      ......
++      ......
++      -----END CERTIFICATE-----
++      "master-0.crt": |
++      -----BEGIN CERTIFICATE-----
++      ......
++      ......
++      ......
++      -----END CERTIFICATE-----
++      ......
+     ......
+     hosts:
+         - name: master-0
+           bmc:
+             disableCertificateVerification: false
+         ......
+     ......
+```
+
+Please note that these CA certificates will be merged into a single file within
+the Ironic container, rather than organized in the form defined by the API.
+Using a map for this field, as opposed to a string, improves maintainability.
+For instance, when updating CA certificates for certain nodes, users can
+distinguish them by the names they have set, reducing the risk of human errors.
+
+### Risks and Mitigations
+
+Users may incorrectly configure the CA certificates, causing the IPI
+installation to timeout. It may take some time for users to realize that the
+failure is due to incorrect certificate configuration.
+
+To mitigate this, we can perform a pre-validation of the SSL connection with
+the BMCs using the provided CA certificate before the installation begins.
+
+### Drawbacks
+
+- Introducing CA certificate support increases complexity to the installation
+  and management process.
+
+- Creating or deleting the CA certificate ConfigMap in an existing cluster
+  causes the Ironic container to restart, leading to temporary unavailability.
+
+- CA certificate pre-validation performed in installer is only for cluster
+  installation. There is no automatic pre-validation process for CA certificate
+  updates in an existing cluster.
+
+## Design Details
+
+### Test Plan
+
+TBD
```

Comment:
This needs populating. We could potentially even tests this on virtual machines since sushy-tools support TLS.

Reply:
Please do update it, we cannot merge the enhancement with literally nothing here.